hydra-node/exe/hydra-node/Main.hs

```diff
@@ -110,6 +126,29 @@ main = do
 
     action (Ledger.cardanoLedger globals ledgerEnv)
 
+  -- check if hydra-node parameters are matching with the hydra-node state.
+  -- REVIEW: Should we also check against items we receive on-chain here? e.g. 'OpenThreadOutput' if in Open state?
```

Comment:
I guess so

Reply:
If we want to check more things I can do a follow up. If this is good already I would just merge it if it's fine.